currency_monthly_rate/models/res_currency.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api
+
+
+class Currency(models.Model):
+    _inherit = "res.currency"
+
+    monthly_rate = fields.Float(compute='_compute_current_monthly_rate',
+                                string='Current Monthly Rate', digits=(12, 6),
+                                help='The monthly rate of the currency to '
+                                     'the currency of rate 1.')
+    monthly_rate_ids = fields.One2many('res.currency.rate.monthly',
+                                       'currency_id', string='Monthly rates')
+
+    @api.multi
+    def _compute_current_monthly_rate(self):
+        date = self._context.get('date') or fields.Date.today()
+        company_id = self._context.get('company_id') or self.env[
+            'res.users']._get_company().id
+        # the subquery selects the first rate before 'date' for the given
+        # currency/company
+        query = """SELECT c.id, (SELECT r.rate
```

Comment:
split this to `_get_currency_rates`

Reply:
The idea was to keep this function as close as possible to this one : https://github.com/odoo/odoo/blob/10.0/odoo/addons/base/res/res_currency.py#L37-L51, so I don't think it's worth it.